src/integrating_GK_affect.jl

```diff
@@ -0,0 +1,167 @@
+ [0.10122853629037629, 0.22238103445337457, 0.3137066458778873, 0.3626837833783619, 0.3626837833783619, 0.3137066458778873, 0.22238103445337457, 0.10122853629037629],
+ [0.08127438836157448, 0.1806481606948575, 0.26061069640293544, 0.31234707704000275, 0.3302393550012595, 0.31234707704000275, 0.26061069640293544, 0.1806481606948575, 0.08127438836157448],
+ [0.06667134430868811, 0.14945134915058084, 0.2190863625159821, 0.26926671930999635, 0.29552422471475276, 0.29552422471475276, 0.26926671930999635, 0.2190863625159821, 0.14945134915058084, 0.06667134430868811]]
+
+
+mutable struct SavingIntegrandGKAffect{
+	IntegrandFunc,
+	tType,
+	IntegrandType,
+	IntegrandCacheType
+}
+	integrand_func::IntegrandFunc
+	integrand_values::IntegrandValues{tType, IntegrandType}
+	integrand_cache::IntegrandCacheType
+	accumulation_cache::IntegrandCacheType
+	gk_step_cache::IntegrandCacheType
+	gk_err_cache::IntegrandCacheType
+	tol::Float64
+end
+
+
+function integrate_gk!(affect!::SavingIntegrandGKAffect, integrator, bound_l, bound_r; order=7, tol=1e-7)
+	affect!.gk_step_cache = recursive_zero!(affect!.gk_step_cache)
+	affect!.gk_err_cache  = recursive_zero!(affect!.gk_err_cache)
+	for i in 1:(2*order+1)					
+		t_temp = (gk_points[order][i]+1)*((bound_r-bound_l)/2) + bound_l 
+		if DiffEqBase.isinplace(integrator.sol.prob)
+			curu = first(get_tmp_cache(integrator))
+			integrator(curu,t_temp)
+			if affect!.integrand_cache == nothing 
+				recursive_axpy!(gk_weights[order][i],
+						affect!.integrand_func(curu, t_temp, integrator),affect!.gk_step_cache)
+				if i%2==0
+					recursive_axpy!(g_weights[order][div(i,2)],
+						affect!.integrand_func(curu, t_temp, integrator),affect!.gk_err_cache)
+				end
+			else
+				affect!.integrand_func(affect!.integrand_cache, curu, t_temp, integrator)
+				recursive_axpy!(gk_weights[order][i],
+						affect!.integrand_cache, affect!.gk_step_cache)
+				if i%2==0
+					recursive_axpy!(g_weights[order][div(i,2)],
+						affect!.integrand_cache, affect!.gk_err_cache)
+				end
+			end
+		else
+			recursive_axpy!(gk_weights[order][i],
+					affect!.integrand_func(integrator(t_temp), t_temp, integrator),affect!.gk_step_cache)		
+			if i%2==0							 
+				recursive_axpy!(g_weights[order][div(i,2)],
+					affect!.integrand_func(integrator(t_temp), t_temp, integrator),affect!.gk_err_cache)
+			end		
+		end
+	end
+	if sum(abs.((affect!.gk_step_cache .- affect!.gk_err_cache).*(bound_r-bound_l)./2))<tol
+		recursive_axpy!(1,affect!.gk_step_cache.*(bound_r-bound_l)./2, affect!.accumulation_cache)
+	else
+		integrate_gk!(affect!, integrator, bound_l, (bound_l+bound_r)/2, order=order, tol=tol/2)
+		integrate_gk!(affect!, integrator, (bound_l+bound_r)/2, bound_r, order=order, tol=tol/2)
+	end
+end
+
+
+function (affect!::SavingIntegrandGKAffect)(integrator)
+    n = 0
+    if integrator.sol.prob isa Union{SDEProblem, RODEProblem}
+        n = 10
```

Comment:
Should error: only trapezoidal is appropriate for this. GK is not necessarily convergent here.